merkle/tree_hasher.go

```diff
@@ -41,60 +41,39 @@ const (
 
 // TreeHasher is a set of domain separated hashers for creating Merkle tree hashes.
 type TreeHasher struct {
-	crypto.Hasher
-	leafHasher  func([]byte) []byte
-	nodeHasher  func([]byte) []byte
-	emptyHasher func() []byte
+	crypto.Hash
 }
 
 // NewRFC6962TreeHasher creates a new TreeHasher based on the passed in hash function.
 // TODO(Martin2112): Move anything CT specific out of here to <handwave> look over there
-func NewRFC6962TreeHasher(hasher crypto.Hasher) TreeHasher {
-	return TreeHasher{
-		Hasher:      hasher,
-		leafHasher:  rfc6962LeafHasher(hasher),
-		nodeHasher:  rfc6962NodeHasher(hasher),
-		emptyHasher: rfc6962EmptyHasher(hasher),
-	}
+func NewRFC6962TreeHasher() TreeHasher {
+	return TreeHasher{Hash: crypto.SHA256}
 }
 
 // HashEmpty returns the hash of an empty element for the tree
 func (t TreeHasher) HashEmpty() []byte {
-	return t.emptyHasher()
+	return t.Digest([]byte{})
 }
 
 // HashLeaf returns the Merkle tree leaf hash of the data passed in through leaf.
 // The data in leaf is prefixed by the LeafHashPrefix.
 func (t TreeHasher) HashLeaf(leaf []byte) []byte {
-	return t.leafHasher(leaf)
+	return t.Digest(append([]byte{RFC6962LeafHashPrefix}, leaf...))
```

Comment:
I understand. This removal of abstractions is temporary.  I'll be replacing the `TreeHasher struct` with a `Hasher interface` in the next PR to recover the full flexibility.